### PR TITLE
Feature/set country code for macos

### PR DESCRIPTION
**Pull request by jinquan2019**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [x] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Add feature: set country code for macOS
1. No need to add the key 'NSLocationWhenInUseUsageDescription' into info.plist because the 'requestWhenInUseAuthorization' method or 'requestAlwaysAuthorization' method is unavailable on macOS 
2. Use the 'startUpdatingLocation' method instead of the 'requestlocation' method because 'requestlocation' method is only available on OSX 10.14+.
3. Add 'stopUpdatingLocation' method when app had fetched the location.

image

For Location service is disabled
image

### CODE REVIEW

File: SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift [hunk 1]
```diff
@@ -1,22 +1,43 @@
...
+    self.locationManager.stopUpdatingLocation()
+    let userLocation:CLLocation = locations[0] as CLLocation
+    CLGeocoder().reverseGeocodeLocation(userLocation) { (placemarks, error) in
+      if error == nil {
+        self.appCenter.setCountryCode(placemarks?.first?.isoCountryCode)
+      }
+    }
+  }
+  
+  func locationManager(_ Manager: CLLocationManager, didFailWithError error: Error){
```
Comment (MatkovIvan) [inline SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift]:
```suggestion
  func locationManager(_ Manager: CLLocationManager, didFailWithError error: Error) {
```

File: SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift [hunk 2]
```diff
@@ -1,22 +1,43 @@
 import Cocoa
+import CoreLocation
 
-class AppCenterViewController : NSViewController {
+class AppCenterViewController : NSViewController, CLLocationManagerDelegate {
 
-  var appCenter: AppCenterDelegate = AppCenterProvider.shared().appCenter!
+  var appCenter: AppCenterDelegate = AppCenterProvider.shared().appCenter!
```
Comment (MatkovIvan) [inline SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift]:
```suggestion
  var appCenter: AppCenterDelegate = AppCenterProvider.shared().appCenter!
```

File: SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift [hunk 3]
```diff
@@ -1,22 +1,43 @@
...
 
-  var appCenter: AppCenterDelegate = AppCenterProvider.shared().appCenter!
+  var appCenter: AppCenterDelegate = AppCenterProvider.shared().appCenter!    
 
   @IBOutlet var installIdLabel : NSTextField?
   @IBOutlet var appSecretLabel : NSTextField?
   @IBOutlet var logURLLabel : NSTextField?
   @IBOutlet var userIdLabel : NSTextField?
   @IBOutlet var setEnabledButton : NSButton?
+  @IBOutlet weak var overrideCountryCodeButton: NSButton!
```
Comment (MatkovIvan) [inline SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift]:
```suggestion
  @IBOutlet var overrideCountryCodeButton: NSButton!
```

**Comment by jaelim-ms** [comment]
As @jwallra mentioned, adding screenshots are strongly recommended for UI PRs.
However, I also suggest all reviewers to checkout the branch and run/test UI implementation if you are blocking/approving UI PRs.

File: SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift [hunk 4]
```diff
@@ -30,4 +51,10 @@ class AppCenterViewController : NSViewController {
     UserDefaults.standard.set(userId, forKey: "userId")
     appCenter.setUserId(userId)
   }
+  
+  @IBAction func overrideCountryCode(_ sender: NSButton) {
+    if CLLocationManager.locationServicesEnabled() {
+      self.locationManager.startUpdatingLocation()
+    }
```
Comment (jaelim-ms) [inline SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift]: We need to inform as a popup dialog that location service isn't enabled. Hard to get noticed if logs don't contain carrier country due to that reason.

**Comment by sachinms27** [comment]
@jinquan2019 I work at Citrix Systems and we are using the AppCenter analytics on one of our macOS Apps. Unavailability of user's geo on macOS was blocking us from using the service. This hook of the App being able to override/set the geo solves our problem. Could you kindly let me know when this commit will be available on the release branch?

**Comment by ElektrojungeMS** [comment]
@sachinms27 If all things are going as expected, it should be part of our March release in the 2nd half of this month.

**Pull request merged**